visualeyes/core/plotting.py

```diff
@@ -1,99 +1,79 @@
+            - for rectangluar AOI's: (x1, x2, y1, y2), upper-bounds non-inclusive. 
+            - for circlular AOI's: (x_center, y_center, radius).
+    - save_png: bool or None
+        If True, the results will be saved as png file(s)
+    - save_path: str or None
+        The file path where the png file(s) will be saved if 'save_png' is True
+        
     Outputs:
-    - ratios: list (returned)
-        ratios of fixations within the Area of Interest (one ratio per trial)
-    - plots: svg
-        plots of fixations ('cross') on the binary image (one svg per trial)
+    - ratios: list
+        ratios of fixations within the Area of Interest (one ratio per epoch)
+    - png file(s): png
 
     """
 
     # check if input data is a pd.DataFrame
-    if not isinstance(dataframe, pd.DataFrame):
-        raise ValueError('input data should be a pandas dataframe')
+    if not isinstance(epoch_data, pd.DataFrame):
+        raise ValueError("Input data should be a pandas dataframe")
     
-    # check if the 'discrete' column has both 'trials' and 'fixations' sub-columns
-    if 'trials' not in dataframe.columns:
-        raise ValueError('missing trials information')
-    if 'fixations' not in dataframe.columns:
-        raise ValueError('missing fixations information')
+    if 'epoch_index' not in epoch_data.columns:
+        raise ValueError("Column 'epoch_index' does not exist in the input dataframe")
 
-    # pull the 'trials' and 'fixations' columns
-    df_trials = dataframe['discrete']['trials']
-    df_fixations = dataframe['discrete']['fixations']
+    # validate the input dataframe and save the x and y coordinates if the dataframe is valid
+    x_coord, y_coord = visualeyes.dataframe_validation(epoch_data)
+    
+    screen_dimension = (aoi_mask.shape[1], aoi_mask.shape[0])
 
-    # start an empty list to save ratio of fixations within the aoi for each trial
-    ratios_fixations_within = []
+    visualeyes.aoi_mask_validation(aoi_mask, screen_dimension)
 
-    for row_idx, trial in df_trials.iterrows():
-        print(f'trial indx: {row_idx}')
-        trial_stime = trial['stime']
-        trial_etime = trial['etime']
+    for index in epoch_data['epoch_index'].unique():
+        epoch_x = x_coord[x_coord['epoch_index'] == index]
+        epoch_y = y_coord[y_coord['epoch_index'] == index]
 
-        # locate the fixations within trial start and end time
-        fixations_in_trial = df_fixations[(df_fixations['stime'] >= trial_stime) & (df_fixations['etime'] <= trial_etime)]
+        out_of_bounds = any(x < 0 or x >= aoi_mask.shape[1] or y < 0 or y >= aoi_mask.shape[0]
+                           for x,y in zip(epoch_x, epoch_y)
+                           )
 
-        # find the x and y coordinates of fixations 
-        x_coor = fixations_in_trial ['axp']
-        y_coor = fixations_in_trial ['ayp']
+        if out_of_bounds:
+            print(f"Some coordinates for epoch {index} are out of bounds")

```

Comment:
`out_of_bounds` returns `True` as long as there's one data point outside screen dimension. Does it make sense to discard all data because of outliers, or is it better to just ignore them?